src/changefile/writeChangeFilesFixup.ts

```diff
@@ -0,0 +1,242 @@
+
+  if (!commitResult.success) {
+    console.error(`Failed to create fixup commit: ${commitResult.errorMessage}`);
+    return null;
+  }
+
+  console.log(`Created fixup commit for ${commitHash}`);
+  console.log(`Updated change file: ${changeFilePath}`);
+
+  return changeFilePath;
+}
+
+const MAX_COMMITS = 100;
+
+/**
+ * Finds the most recent commit that introduced change files by searching through
+ * the current branch's commits from newest to oldest.
+ * @param cwd - Working directory
+ * @param changePath - Path to the change directory
+ * @returns Object with commit hash and change file name, or null if none found
+ */
+async function findMostRecentChangeFileCommit(
+  cwd: string,
+  changePath: string
+): Promise<{ commitHash: string; changeFileName: string } | null> {
+  const relativeChangePath = path.relative(cwd, changePath);
+
+  // Get the fork point (merge base with main/master branch)
+  const forkPointResult = await findForkPoint(cwd);
+  if (!forkPointResult) {
+    console.warn('Unable to determine fork-point for current branch');
+    return null;
+  }
+
+  // Get commits on the current branch, limited to MAX_COMMITS
+  const logResult = await gitAsync(
+    ['log', '--oneline', '--format=%H', `-n`, MAX_COMMITS.toString(), `${forkPointResult}..HEAD`],
+    { cwd }
+  );
+  if (!logResult.success) {
+    console.warn('Failed to get git log for finding recent change file commits');
+    return null;
+  }
+
+  const commits = logResult.stdout
+    .trim()
+    .split('\n')
+    .filter(line => line.trim());
+
+  // Search through commits from newest to oldest
+  for (const commitHash of commits) {
+    // Check what files were added in this commit (compare with parent)
+    const diffResult = await gitAsync(
+      [
+        'diff-tree',
+        '--no-commit-id',
+        '--name-status',
+        '--diff-filter=A', // Only look for added files
+        '-r', // Recursive to show files inside directories
+        commitHash,
+      ],
+      { cwd }
+    );
+
+    if (!diffResult.success) {
+      continue;
+    }
+
+    // Look for added change files
+    const addedChangeFiles: string[] = [];
+    const lines = diffResult.stdout
+      .trim()
+      .split('\n')
+      .filter(line => line.trim());
+
+    for (const line of lines) {
+      // Handle both formats: "A\tfilepath" and "filepath" (when only added files are shown)
+      const parts = line.split('\t');
+      const filePath = parts.length > 1 ? parts[1] : line;
+
+      if (filePath) {
+        // Normalize paths for comparison
+        const normalizedPath = filePath.replace(/\\/g, '/');
+        const normalizedChangePath = relativeChangePath.replace(/\\/g, '/');
+
+        if (normalizedPath.startsWith(normalizedChangePath) && normalizedPath.endsWith('.json')) {
+          const fileName = path.basename(normalizedPath);
+          addedChangeFiles.push(fileName);
+        }
+      }
+    }
+
+    if (addedChangeFiles.length > 0) {
```

Comment:
It seems like this will only work properly with grouped change files? For ungrouped change files, you either need to find the right file for the specific package, or just create a new file and add it to the commit with change files. Creating a new file is most similar to what you're doing with grouped changes.